tests/new_core/processors/test_processor_utils.py

```diff
@@ -0,0 +1,1444 @@
+        )
+
+        assert isinstance(result, float)
+        assert result >= 0
+
+    def test_large_dataset_approximation(self):
+        """Test ESS calculation with large dataset approximation."""
+        # Create a large gridded dataset that triggers approximation logic
+        large_data = TestDataFactory.create_climate_dataset(
+            time_periods=10000,  # Large enough to trigger approximation
+            lat_points=3,
+            lon_points=3,
+        ).rename({"lat": "x", "lon": "y"})
+
+        result = _calc_average_ess_gridded_optimized(large_data, block_size=1)
+
+        assert isinstance(result, (int, float))
+        assert not np.isnan(result)
+
+    @pytest.mark.advanced
+    def test_dask_array_ess_calculation(self):
+        """Test ESS calculation with Dask arrays."""
+        # Create dask-backed dataset
+        dask_data = TestDataFactory.create_climate_dataset(
+            time_periods=1000, lat_points=5, lon_points=5, with_dask=True
+        ).rename({"lat": "x", "lon": "y"})
+
+        result = _calc_average_ess_gridded_optimized(dask_data, block_size=1)
+
+        assert isinstance(result, (int, float))
+        assert not np.isnan(result)
+
+    @patch("climakitae.explore.threshold_tools.calculate_ess")
+    def test_ess_calculation_with_errors(self, mock_calculate_ess):
+        """Test ESS calculation when calculate_ess raises errors."""
+        # Mock to raise exceptions then return valid results
+        mock_calculate_ess.side_effect = [
+            ValueError("Error"),
+            xr.DataArray([25.0]),
+            RuntimeError("Runtime error"),
+            xr.DataArray([30.0]),
+        ]
+
+        # Create dataset that will trigger the error handling
+        error_data = TestDataFactory.create_climate_dataset(
+            time_periods=365 * 5, lat_points=2, lon_points=2
+        ).rename({"lat": "x", "lon": "y"})
+
+        result = _calc_average_ess_gridded_optimized(error_data, block_size=1)
+
+        # Should handle errors and return average of successful calculations
+        assert isinstance(result, (int, float))
+        assert not np.isnan(result)
+
+    def test_ess_calculation_memory_error(self):
+        """Test ESS calculation when MemoryError occurs."""
+        # Create a scenario that might cause memory issues by mocking calculate_ess
+        with patch(
+            "climakitae.explore.threshold_tools.calculate_ess",
+            side_effect=MemoryError("Out of memory"),
+        ):
+            error_data = TestDataFactory.create_climate_dataset(
+                time_periods=365 * 2, lat_points=2, lon_points=2
+            ).rename({"lat": "x", "lon": "y"})
+
+            result = _calc_average_ess_gridded_optimized(error_data, block_size=1)
+
+            # Should return a reasonable ESS value or fallback
+            assert isinstance(result, (int, float))
+            assert result > 0  # ESS should be positive
+
+    def test_ess_calculation_large_dataset(self):
```

Comment:
This test isn't getting into `if n > LARGE_DATA_THRESHOLD` codeblock in `processor_utils.py`